src/run.js

```diff
@@ -211,8 +219,9 @@ const minimalcss = async options => {
         waitUntil: ['domcontentloaded', 'networkidle2']
       })
       if (!response.ok) {
-        return reject(
-          new Error(`${response.status} on ${pageUrl} (second time)`)
+        return (
+          !pageerrorCalled &&
+          reject(new Error(`${response.status} on ${pageUrl} (second time)`))
```

Comment:
Can you move this, and the block on line 194-197 to a function? Perhaps something like:
```javascript
if (!response.ok) {
  return rejectResponse(response, 'second time')
}
```
and then that function could be something like 
```javascript
const rejectResponse = (response, message) => {
  if (!pageerrorCalled) {
     // We have not previously errored out before, so do it for this bad response
    const error = new Error(`${response.status} on ${pageUrl} (${message})`)
    reject(error)
  }
}
```